src/targets/gpu/include/migraphx/gpu/gemm_impl.hpp

```diff
@@ -24,28 +24,492 @@
+    {
+        std::vector<argument> input_args;
+        std::transform(input_shapes.begin(),
+                       input_shapes.end(),
+                       std::back_inserter(input_args),
+                       [](const shape& x) { return to_gpu(generate_argument(x)); });
+
+        // Find out how many solutions there are
+        rocblas_int list_size = 0;
+        std::vector<rocblas_int> solution_indices;
+        if(strided_batched)
+        {
+            auto arg_list = create_strided_batched_gemm_get_solutions_args(
+                ctx, input_args, nullptr, &list_size);
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions, arg_list);
+            solution_indices.resize(list_size);
+            auto arg_list_solutions = create_strided_batched_gemm_get_solutions_args(
+                ctx, input_args, solution_indices.data(), &list_size);
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions, arg_list_solutions);
+        }
+        else
+        {
+            auto arg_list = create_gemm_ex_get_solutions_args(ctx, input_args, nullptr, &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions, arg_list);
+            solution_indices.resize(list_size);
+            auto arg_list_solutions = create_gemm_ex_get_solutions_args(
+                ctx, input_args, solution_indices.data(), &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions, arg_list_solutions);
+        }
+
+        double bestTime = std::numeric_limits<double>::max();
+        // Initialize to default solution index
+        rocblas_int bestSol = 0;
+        for(auto sol : solution_indices)
+        {
+            // Warmup: the first few calls to an op. may not be representative since there is
+            // more time taken initializing caches, etc. so we won't time them.
+            for(rocblas_int cc = 0; cc < cold_calls; ++cc)
+            {
+                run(ctx, input_args, sol);
+            }
+            double host_time = 0.0;
+            // Define the function to be timed
+            auto run_func    = [&]() {
+                run(ctx, input_args, sol);
+                ctx.finish();
+            };
+            for(rocblas_int hc = 0; hc < hot_calls; ++hc)
+            {
+                ctx.finish();
+                host_time += time<microseconds>(run_func);
+            }
+            // todo:  Measured time dropped from 20 us to about 6.7 us when I raised hot_calls from
+            // 1 to 11. The higher the hot_calls value, the faster per-call time up to at least 25,
+            // and increasing cold_calls makes little or no difference.  Why?
+            host_time /= hot_calls;
+            std::cout << "Solution index " << sol << ": " << host_time << " us" << std::endl;
+
+            // track current best
+            if(host_time < bestTime)
+            {
+                printf(" current best index  %d\n", sol);
```

Comment:
This should use `std::cerr` or `std::clog` instead of plain `printf`.

Reply:
debug code--I don't think we're going to want that line in the delivered product at all.